warp10/src/main/java/io/warp10/script/functions/ALMOSTEQ.java

```diff
@@ -40,26 +39,22 @@ public ALMOSTEQ(String name) {
   @Override
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
     Object lambdaParam = stack.pop();
-    Object op2 = stack.pop();
-    Object op1 = stack.pop();
-    
-    double lambda;
+    Object op2Param = stack.pop();
+    Object op1Param = stack.pop();
 
-    if (lambdaParam instanceof Number) {
-      lambda = Math.abs(((Number) lambdaParam).doubleValue());
-    } else {
-      throw new WarpScriptException(getName() + " needs three parameters on the stack: 'lambda' (a numeric type) and two operands.");  
+    // All operands need to be numeric.
+    if  (!(lambdaParam instanceof Number) || !(op2Param instanceof Number)  || !(op1Param instanceof Number)) {
+      throw new WarpScriptException(getName() + " only accepts numeric parameters.");
     }
 
-    // At least one of the operands must be a Double and both operands need to be numeric types
-    if  (!(op2 instanceof Number)  || !(op1 instanceof Number) || !(op2 instanceof Double || op1 instanceof Double)) {
-      throw new WarpScriptException(getName() + " can only operate on numeric types and at least one of them must be a Double.");  
-    }
-    
-    if (Double.isNaN(((Number) op1).doubleValue()) || Double.isNaN(((Number) op2).doubleValue())) {
-      stack.push(Double.isNaN(((Number) op1).doubleValue()) && Double.isNaN(((Number) op2).doubleValue()));
+    double lambda = Math.abs(((Number) lambdaParam).doubleValue());
+    double op1 = Math.abs(((Number) op1Param).doubleValue());
```

Comment:
Why use Math.abs for computing values of op1 and op2?

Reply:
My bad, fixed.